homework_4.03_android_components

AliakseiShvants opened the pull request:
Created simple fragment, added service and broadcast receiver

AliakseiShvants committed:
add toast

AliakseiShvants committed:
add simple fragment and use saving fragment state

AliakseiShvants committed:
add service and broadcast receiver

AliakseiShvants committed:
removed .idea & gradle from git

AliakseiShvants committed:
Merge branch 'master' into hw_4.03

ajaxspace commented on app/src/main/AndroidManifest.xml:
```diff
@@ -8,7 +9,17 @@
+
+        <receiver android:name=".CounterBroadcastReceiver">
+            <intent-filter>
+                <action android:name="com.shvants.CUSTOM_ACION" />
```
Read about https://developer.android.com/studio/build/manifest-build-variables.

AliakseiShvants replied:
applied 'applicationId' https://github.com/AliakseiShvants/RunningLife/compare/01842e8f4dd29c265bdc458cc8a8a71b6eae8310...58e65bc2b5d572145ec44afbcbc683ee2b932035

ajaxspace commented on app/src/main/java/com/shvants/runninglife/MainActivity.java:
```diff
@@ -1,13 +1,35 @@
 
 public class MainActivity extends AppCompatActivity {
 
+    public static final String RESULT = "RESULT";
```
There is already such field in CounterBroadcastReceiver

AliakseiShvants replied:
moved this constant in 'Const' class https://github.com/AliakseiShvants/RunningLife/compare/01842e8f4dd29c265bdc458cc8a8a71b6eae8310...58e65bc2b5d572145ec44afbcbc683ee2b932035

ajaxspace commented on app/src/main/java/com/shvants/runninglife/MainActivity.java:
```diff
@@ -26,5 +47,63 @@ public void onClick(final View v) {
+
+        findViewById(R.id.main_stop_service_btn).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View v) {
```
Calling this logic if service is not started will lead to crash.

AliakseiShvants replied:
@ajaxspace made stop button not clickable until start button is pressed https://github.com/AliakseiShvants/RunningLife/compare/01842e8f4dd29c265bdc458cc8a8a71b6eae8310...58e65bc2b5d572145ec44afbcbc683ee2b932035

AliakseiShvants committed:
move 'RESULT' constant in 'Const' class;
make stop button not clickable until start button will be pressed
add 'applicationId' in AndroidManifest
